src/discovery/discovery.utils.ts

```diff
@@ -0,0 +1,314 @@
+const SHA3_NULL_S =
+  '0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470'; //TODO: verify this is the correct way to get the hash. Copied from rif-relay-common (https://github.com/rsksmart/rif-relay-common/blob/f8e4ce8d90a979f3eec55a0eb5dbd5cc742c40e3/src/Constants.ts#L18-L19)
+
+type Account = {
+  eoaAccount: string;
+  swAccounts: string[];
+};
+
+// TODO: copied from @ethersproject/wordlists/src.ts/wordlists.ts as that doesn't allow for locale type extraction. Remove (including imports) when this is fixed (https://github.com/ethers-io/ethers.js/pull/3514).
+const wordlists = {
+  cz,
+  en,
+  es,
+  fr,
+  it,
+  ja,
+  ko,
+  zh: zh_cn,
+  zh_cn,
+  zh_tw,
+} as const;
+
+type Locale = keyof typeof wordlists; //TODO: hopefully this will be added to ethers.js (https://github.com/ethers-io/ethers.js/pull/3514)
+
+// TODO: check that the optional params have to be optional
+type Config = {
+  eoaGap: number;
+  sWalletGap: number;
+  searchBalance: boolean;
+  searchNonce: boolean;
+  searchTokenBalance: boolean;
+  searchDeployEvents: boolean;
+  network: string; //TODO: potentially restictable by program-wide config
+  factory: string;
+  isCustomWallet: boolean;
+  mnemonicLanguage: Locale;
+  recoverer: string;
+  logic: string;
+  logicParamsHash: string;
+  reconnectionAttempts: number;
+  tokens: string[];
+};
+
+const DEFAULT_DISCOVERY_CONFIG: Partial<Config> = {
+  eoaGap: 20,
+  sWalletGap: 20,
+  searchBalance: true,
+  searchNonce: true,
+  searchTokenBalance: true,
+  searchDeployEvents: true,
+  network: 'mainnet', //FIXME: this should probably be something else. Also should be linked/mapped to the network id or address defined in program-wide config or env vars;
+  mnemonicLanguage: 'en',
+  recoverer: constants.AddressZero,
+  tokens: [],
+};
+
+//Account discovery https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#account-discovery
+const DEFAULT_DERIVATION_PATH_PURPOSE = 44;
+const RSK_COIN_TYPE = 137; // src: https://github.com/satoshilabs/slips/blob/master/slip-0044.md
+const RSK_TESTNET_COIN_TYPE = 37310; // src: https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP57.md
+const DEFAULT_DERIVATION_PATH_CHANGE = 0;
+
+const constructDerivationPath = (
+  addressIdx: number,
+  hardenedIdx: number,
+  coinType?: number,
+  isTestNet = false,
+  purpose = DEFAULT_DERIVATION_PATH_PURPOSE,
+  change: 0 | 1 = DEFAULT_DERIVATION_PATH_CHANGE
+) => {
+  return `m/${purpose}'/${
+    coinType || isTestNet ? RSK_TESTNET_COIN_TYPE : RSK_COIN_TYPE
+  }'/${hardenedIdx}'/${change}/${addressIdx}`;
+};
+
+const getSWAddress = (
+  config: Config,
+  ownerEOA: string,
+  walletIndex: number,
+  bytecodeHash: string
+) => {
+  const isCustom = config.logic && config.logicParamsHash;
+  const salt = isCustom
+    ? solidityKeccak256(
+        ['address', 'address', 'address', 'bytes32', 'uint256'],
+        [
+          ownerEOA,
+          config.recoverer,
```

Comment:
Why is this information coming from the configuration? This should be a param not a configuration.